rskj-core/src/main/java/org/ethereum/vm/trace/ProgramTraceProcessor.java

```diff
@@ -20,25 +20,34 @@
 package org.ethereum.vm.trace;
 
 import co.rsk.crypto.Keccak256;
+import co.rsk.rpc.modules.debug.TraceOptions;
 import com.fasterxml.jackson.annotation.JsonAutoDetect;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ser.impl.SimpleBeanPropertyFilter;
+import com.fasterxml.jackson.databind.ser.impl.SimpleFilterProvider;
 
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Objects;
+import java.util.*;
 import java.util.stream.Collectors;
 
 /**
  * Provides tracing and exporting to JSON
  */
 public class ProgramTraceProcessor {
 
-    private static final ObjectMapper OBJECT_MAPPER = makeObjectMapper();
-
     private final Map<Keccak256, ProgramTrace> traces = new HashMap<>();
 
+    private TraceOptions traceOptions;
+
+    public ProgramTraceProcessor() {
+        traceOptions = new TraceOptions();
+    }
+
+    public ProgramTraceProcessor(TraceOptions options) {
+        this();
```

Comment:
do we need this() call since we will overwrite `traceOptions` in the next line?

Reply:
You're completely right! It remained like that after a refactoring was made. Solved!